src/backend/bs_worker

```diff
@@ -3075,7 +3075,24 @@ sub dobuild {
 
   writestr("$buildroot/.build.meta", undef, join("\n", @meta)."\n");
 
-  getoldpackages($buildinfo, $oldpkgdir) if $oldpkgdir && !$kiwimode && $buildinfo->{'file'} ne 'preinstallimage';
+  my @configpath;
+  if ($kiwimode) {
+    @configpath = map {"path=$_->{'project'}/$_->{'repository'}"} @{$buildinfo->{'syspath'} || $buildinfo->{'path'} || []};
+    unshift @configpath, "path=$projid/$repoid" unless @configpath && $configpath[0] eq "path=$projid/$repoid";
+  }
+  my $server = $buildinfo->{'srcserver'} || $srcserver;
+  my $config = BSRPC::rpc("$server/getconfig", undef, "project=$projid", "repository=$repoid", @configpath);
+  writestr("$buildroot/.build.config", undef, $config);
+
+
+  my $bconf = Build::read_config($buildinfo->{'arch'}, "$buildroot/.build.config");
+  my $useccache=0;
+  if (exists $bconf->{'buildflags:useccache'}) {
+      if (grep /^useccache:\Q$packid\E$/, @{$bconf->{'buildflags'} || []}) {
```

Comment:
Having said the above for _ccache.tar.gz, this could also be
```
if (grep {$_ eq "useccache:$packid"} @{$bconf->{'buildflags'} || []}) {
```
Just saying...

Oh, and should we rename the flag to `pkgccache` to match the build option? Just a thought...